packages/frontend/src/contexts/signUps.js

```diff
@@ -0,0 +1,41 @@
+import { makeAutoObservable } from 'mobx'
+import { SERVER } from '../config'
+
+export default class SignUps {
+  allSignUps = []
+  attesterSignUps = []
+  userSignUps = []
+
+  constructor(state) {
+    makeAutoObservable(this)
+    this.state = state
+    if (typeof window !== 'undefined') {
+      //   this.load()
+    }
+  }
+
+  //   async load() {}
+
+  async loadAllSignUps() {
+    const url = new URL(`api/signups`, SERVER)
+    const response = await fetch(url.toString()).then((r) => r.json())
+    console.log('..AllSignups was called')
+    console.log('response: ', response)
+    this.allSignUps = response
+    console.log('extracted list:', this.allSignUps)
+  }
+
```

Comment:
Opted to delete all previous routes for a fresh start. Currently have 3 routes to `UserSignUp` table (for all, byAttester, byUser), all 3 called from single `signUps` context. I've called each function from `HomePage` for my initial understanding/debugging.

<img width="1151" alt="Screen Shot 2023-01-13 at 12 02 48 PM" src="https://user-images.githubusercontent.com/79813968/212398512-a09603f2-4c5a-47f7-b4fd-baede256d41c.png">
... this is the same issue I was having with the response in my previous notes... I've spent 2 days trying to assign the array from 'response' to an array `allSignUps`. Every approach returns `allSignUps` as a proxy. So I am trying to understand some things about proxies, but I feel that I must be missing something more obvious here and shouldn't need to do that.

Reply:
The proxy object is a wrapper that mobx uses. Basically each javascript object becomes a proxy so that when you change the object and code that is observing it can be updated.

If you look at the `target` data in the proxy you can see the underlying data.